File: src/main/java/dev/lumberisland/core/events/PlayerBreakTreeEvent.java
package dev.lumberisland.core.events;

import dev.lumberisland.core.events.system.PluginEvent;
import dev.lumberisland.core.objects.Tree;
import org.bukkit.entity.Player;

import java.util.HashSet;
import java.util.Set;
import java.util.function.Consumer;

public class PlayerBreakTreeEvent implements PluginEvent<PlayerBreakTreeEvent> {

    private static final Set<Consumer<PlayerBreakTreeEvent>> events = new HashSet<>();

    private final Player player;

    public Player getPlayer() {
        return player;
    }

    public Tree getTree() {
        return tree;
    }

    private final Tree tree;

    public PlayerBreakTreeEvent(Player player, Tree tree) {
        this.player = player;
        this.tree = tree;
    }

    @Override
    public PlayerBreakTreeEvent getEvent() {
        return this;
    }

    @Override
    public Set<Consumer<PlayerBreakTreeEvent>> getEventList() {
        return events;
    }
}

File: src/main/java/dev/lumberisland/core/events/PlayerBreakTreeEvent.kt
package dev.lumberisland.core.events

import dev.lumberisland.core.events.system.PluginEvent
import dev.lumberisland.core.events.system.stereotype.EventHandler
import dev.lumberisland.core.events.system.stereotype.EventLimiter
import dev.lumberisland.core.objects.Tree
import org.bukkit.entity.Player
import java.util.function.Consumer

class PlayerBreakTreeEvent(val player: Player, val tree: Tree) : PluginEvent<PlayerBreakTreeEvent> {

    override fun getEvent(): PlayerBreakTreeEvent {
        return this
    }

    override fun getEventList(): MutableSet<Consumer<PlayerBreakTreeEvent>> {
        return events
    }

    companion object {
        @EventHandler
        @EventLimiter(size = 5)
        private val events: MutableSet<Consumer<PlayerBreakTreeEvent>> = HashSet()
    }
}